### PR TITLE
Automated cherry pick of #71681: Add e2e test for file exec

### PR DESCRIPTION
**Pull request by verult**
Cherry pick of #71681 on release-1.11.

#71681: Add e2e test for file exec

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/storage/volumes.go [hunk 1]
```diff
@@ -119,6 +124,20 @@ var _ = utils.SIGDescribe("Volumes", func() {
...
+
+		It("should allow exec of files on the volume", func() {
+			config, _, serverIP := framework.NewNFSServer(cs, namespace.Name, []string{})
+			defer framework.VolumeTestCleanup(f, config)
+
+			volume := &v1.VolumeSource{
+				NFS: &v1.NFSVolumeSource{
+					Server:   serverIP,
+					Path:     "/",
+					ReadOnly: true,
```
Comment (msau42) [inline test/e2e/storage/volumes.go]: By setting readOnly, I think we won't be able to echo the script to exec

**Comment by verult** [comment]
This is adding a new test case for a bug found with certain kernel versions.

/assign @foxish

**Comment by verult** [comment]
Addressed comments

**Comment by foxish** [comment]
@msau42, is this good to go?

**Comment by verult** [comment]
Fixed the gofmt error, but I'm not sure where the bazel build error came from. Maybe a flake?

@msau42 btw I changed the test image from Nginx to NginxSlim (since the former isn't defined in 1.11). Also simplified the test function a bit by removing the bits tailored to the new e2e testing framework

**Comment by verult** [comment]
@foxish good to go!

**Pull request merged**